panic: Non-nullable column with no value!

Running v2.1.0-beta.20180827 on a 5 nodes cluster. One of the node crash with this panic message.

The column screenview:lastmodified was defined with default value.

lastmodified TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT now()

```
panic: Non-nullable column "screenview:lastmodified" with no value! Index scanned was "primary" with the index key columns (fp) and the values ('a6b301a0-1e78-4263-b351-e1347c99c2fa')

goroutine 160961941 [running]:
github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*RowFetcher).finalizeRow(0xc42687fc90, 0x2efbe40, 0xc43a67a0c0)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/rowfetcher.go:1276 +0x9d9
github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*RowFetcher).NextRow(0xc42687fc90, 0x2efbe40, 0xc43a67a0c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/rowfetcher.go:1028 +0x183
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*rowFetcherWrapper).Next(0xc430304ca0, 0xc42b1efb30, 0x0, 0x0, 0xc42b1efb30)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/tablereader.go:141 +0x4a
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*tableReader).Next(0xc42687f800, 0x0, 0x0, 0x0, 0xc42b1efb30)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/tablereader.go:238 +0x4e
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.Run(0x2efbe40, 0xc446d8aba0, 0x2f06300, 0xc42687f800, 0x2eda000, 0xc432698000)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go:170 +0x35
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*ProcessorBase).Run(0xc42687f800, 0x2efbe40, 0xc446d8aba0, 0xc42be2ff38)
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/processors.go:739 +0x98
created by github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).StartAsync
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow.go:574 +0x1e3
```

```
I180830 18:13:33.371469 537 server/status/runtime.go:433  [n4] runtime stats: 10 GiB RSS, 360 goroutines, 238 MiB/154 MiB/562 MiB GO alloc/idle/total, 7.9 GiB/9.9 GiB CGO alloc/total, 199311.67cgo/sec, 1.04/0.05 %(u/s)time, 0.00 %gc (3x)
I180830 18:13:34.451296 539 server/status/recorder.go:652  [n4,summaries] available memory from cgroups (8.0 EiB) exceeds system memory 30 GiB, using system memory
I180830 18:13:36.358093 160915042 sql/distsqlrun/backfiller.go:226  [n4,IndexBackfiller=90] retrying adding checkpoint /Table/90/1/190{744971/16064/0-863091/1491} to table t_demographic_survey
I180830 18:13:43.374066 537 server/status/runtime.go:433  [n4] runtime stats: 10 GiB RSS, 345 goroutines, 217 MiB/176 MiB/576 MiB GO alloc/idle/total, 7.9 GiB/9.9 GiB CGO alloc/total, 66491.83cgo/sec, 0.73/0.06 %(u/s)time, 0.00 %gc (2x)
I180830 18:13:44.451265 539 server/status/recorder.go:652  [n4,summaries] available memory from cgroups (8.0 EiB) exceeds system memory 30 GiB, using system memory
I180830 18:13:53.372411 537 server/status/runtime.go:433  [n4] runtime stats: 10 GiB RSS, 293 goroutines, 214 MiB/181 MiB/577 MiB GO alloc/idle/total, 7.9 GiB/9.9 GiB CGO alloc/total, 174182.79cgo/sec, 0.92/0.05 %(u/s)time, 0.00 %gc (2x)
I180830 18:13:54.451300 539 server/status/recorder.go:652  [n4,summaries] available memory from cgroups (8.0 EiB) exceeds system memory 30 GiB, using system memory
I180830 18:13:57.312758 160933402 sql/distsqlrun/backfiller.go:226  [n4,IndexBackfiller=90] retrying adding checkpoint /Table/90/1/190{749311/14696/0-863091/1491} to table t_demographic_survey
I180830 18:14:03.372848 537 server/status/runtime.go:433  [n4] runtime stats: 10 GiB RSS, 273 goroutines, 183 MiB/201 MiB/580 MiB GO alloc/idle/total, 7.9 GiB/9.9 GiB CGO alloc/total, 2959.47cgo/sec, 0.60/0.06 %(u/s)time, 0.00 %gc (2x)
I180830 18:14:04.451217 539 server/status/recorder.go:652  [n4,summaries] available memory from cgroups (8.0 EiB) exceeds system memory 30 GiB, using system memory
I180830 18:14:07.663524 160942899 sql/distsqlrun/backfiller.go:226  [n4,IndexBackfiller=90] retrying adding checkpoint /Table/90/1/190{751541/14676/0-863091/1491} to table t_demographic_survey
I180830 18:14:07.779057 160942899 sql/distsqlrun/backfiller.go:226  [n4,IndexBackfiller=90] retrying adding checkpoint /Table/90/1/190{751541/14676/0-863091/1491} to table t_demographic_survey
I180830 18:14:13.372046 537 server/status/runtime.go:433  [n4] runtime stats: 10 GiB RSS, 262 goroutines, 331 MiB/120 MiB/580 MiB GO alloc/idle/total, 7.9 GiB/9.9 GiB CGO alloc/total, 90983.63cgo/sec, 0.73/0.06 %(u/s)time, 0.00 %gc (1x)
I180830 18:14:14.451285 539 server/status/recorder.go:652  [n4,summaries] available memory from cgroups (8.0 EiB) exceeds system memory 30 GiB, using system memory
panic: Non-nullable column "screenview:lastmidifed" with no value! Index scanned was "primary" with the index key columns (fp) and the values ('a6b301a0-1e78-4263-b351-e1347c99c2fa')

goroutine 160961941 [running]:
github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*RowFetcher).finalizeRow(0xc42687fc90, 0x2efbe40, 0xc43a67a0c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/rowfetcher.go:1276 +0x9d9
github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*RowFetcher).NextRow(0xc42687fc90, 0x2efbe40, 0xc43a67a0c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/rowfetcher.go:1028 +0x183
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*rowFetcherWrapper).Next(0xc430304ca0, 0xc42b1efb30, 0x0, 0x0, 0xc42b1efb30)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/tablereader.go:141 +0x4a
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*tableReader).Next(0xc42687f800, 0x0, 0x0, 0x0, 0xc42b1efb30)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/tablereader.go:238 +0x4e
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.Run(0x2efbe40, 0xc446d8aba0, 0x2f06300, 0xc42687f800, 0x2eda000, 0xc432698000)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/base.go:170 +0x35
github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*ProcessorBase).Run(0xc42687f800, 0x2efbe40, 0xc446d8aba0, 0xc42be2ff38)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/processors.go:739 +0x98
created by github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).StartAsync
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow.go:574 +0x1e3

```

I'm now betting that this crash is happening in the index backfiller. The index backfiller runs through a scan of the table and it hits this panic. Normally panics caused by statements result in the panic showing up logging an anonymized version of the statement.

I can replicate this panic by running the following query when the concurrent insert is happening.

select * from universe.screenview where pp= '129d5825-f5b9-4535-bcad-5aaf2ad85bba'

Where '129d5825-f5b9-4535-bcad-5aaf2ad85bba' is exists and the panic message will be for another currently insert row.

panic: Non-nullable column "screenview:lastmodified" with no value! Index scanned was "primary" with the index key columns (fp) and the values ('563e10ba-1a4c-4416-81aa-dcf9dcbb6662')

--- a/pkg/sql/distsqlrun/distinct.go
+++ b/pkg/sql/distsqlrun/distinct.go
@@ -215,6 +215,9 @@ func (d *Distinct) Next() (sqlbase.EncDatumRow, *ProducerMetadata) {
 	for d.State == StateRunning {
 		row, meta := d.input.Next()
 		if meta != nil {
+			if meta.Err != nil {
+				d.MoveToDraining(nil /* err */)
+			}
 			return nil, meta
 		}
 		if row == nil {
@@ -282,6 +285,9 @@ func (d *SortedDistinct) Next() (sqlbase.EncDatumRow, *ProducerMetadata) {
 	for d.State == StateRunning {
 		row, meta := d.input.Next()
 		if meta != nil {
+			if meta.Err != nil {
+				d.MoveToDraining(nil /* err */)
+			}
 			return nil, meta
 		}
 		if row == nil {

--- a/pkg/sql/distsqlrun/sorter.go
+++ b/pkg/sql/distsqlrun/sorter.go
@@ -406,6 +406,10 @@ func (s *sortTopKProcessor) Start(ctx context.Context) context.Context {
 		row, meta := s.input.Next()
 		if meta != nil {
 			s.trailingMeta = append(s.trailingMeta, *meta)
+			if meta.Err != nil {
+				s.MoveToDraining(nil /* err */)
+				break
+			}
 			continue
 		}
 		if row == nil {
@@ -549,6 +553,9 @@ func (s *sortChunksProcessor) fill() (bool, error) {
 
 		if meta != nil {
 			s.trailingMeta = append(s.trailingMeta, *meta)
+			if meta.Err != nil {
+				return false, nil
+			}
 			continue
 		}
 		if nextChunkRow == nil {

--- a/pkg/sql/distsqlrun/tablereader.go
+++ b/pkg/sql/distsqlrun/tablereader.go
@@ -242,6 +242,9 @@ func (tr *tableReader) Next() (sqlbase.EncDatumRow, *ProducerMetadata) {
 		row, meta := tr.input.Next()
 
 		if meta != nil {
+			if meta.Err != nil {
+				tr.MoveToDraining(nil /* err */)
+			}
 			return nil, meta
 		}
 		if row == nil {

